libr/cons/dietline.c

```diff
@@ -1593,7 +1593,7 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 					buf[1] = r_cons_readchar_timeout (50);
 					if (buf[1] == -1) {
 						r_cons_break_pop ();
-						return NULL;
+						return I.buffer.data;
```

Comment:
I'm not sure this is the right fix. Why shall we return something here, but not in e.g. https://github.com/radareorg/radare2/pull/17513/files#diff-1b276f78ec258bb5c8baaa8447300b2dL1607 or https://github.com/radareorg/radare2/pull/17513/files#diff-1b276f78ec258bb5c8baaa8447300b2dL1668 .

From what I can see the issue seems to be in `core.c:r_core_prompt_loop` and/or `core.c:r_core_prompt`. Indeed in `r_core_prompt` the function returns `false`(aka 0) when `ret == -1` and it returns `R_CORE_CMD_EXIT` when `ret == -2`. Now, `r_core_prompt_loop` should exit from the prompt loop *only* when `err == R_CORE_CMD_EXIT`, but the current check is ` if (err < 1 ) // handle ^D .... ; break;` so also the case where `ret` is `false` is considered the same as `R_CORE_CMD_EXIT`.

Reply:
this returns the current buffer instead of NULL which was causing the program to be closed

Reply:
I'm not sure I understand your reply. I do know you are trying to fix the issue of radare2 being closed, but as I explained in the comment above, I do not think this is the right solution (again, for the reasons explained above).

Reply:
Yes, that breaks the line and we probably just want to get rid of the trashy chars and continue in the read loop. ill look into that